.github/workflows/aws-staging-kubernetes.yaml

```diff
@@ -0,0 +1,110 @@
+on:
+  push:
+    branches:
+      - dev-k8s
+
+name: Deploy to AWS Staging (K8s)
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment: staging
+    env:
+      AWS_REGION: eu-central-1
+      CLUSTER_NAME: maker-staging
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ env.AWS_REGION }}
+
+    - name: Login to Amazon ECR
+      id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v1
+
+    - name: Extract commit hash
+      id: vars
+      shell: bash
+      run: |
+        echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+    - name: Build, tag, and push image to Amazon ECR
+      id: build-image
+      env:
+        SHA_TAG: ${{ steps.vars.outputs.sha_short }}
+        LATEST_TAG: latest
+        ENVIRONMENT_TAG: staging
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      run: |  
+        # Build Docker containers and
+        # push them to ECR:
+        # - chainlog-ui
+        docker pull $ECR_REGISTRY/chainlog-ui-staging:$LATEST_TAG || true
+        docker build -t chainlog-ui-staging \
+                     -t $ECR_REGISTRY/chainlog-ui-staging:$SHA_TAG \
+                     -t $ECR_REGISTRY/chainlog-ui-staging:$LATEST_TAG \
+                     -t $ECR_REGISTRY/chainlog-ui-staging:$ENVIRONMENT_TAG \
+                     -f Dockerfile.ui \
+                     .
+        docker push $ECR_REGISTRY/chainlog-ui-staging --all-tags
+        # - chainlog-checksum
+        docker pull $ECR_REGISTRY/chainlog-checksum-staging:$LATEST_TAG || true
+        docker build -t chainlog-checksum-staging \
+                     -t $ECR_REGISTRY/chainlog-checksum-staging:$SHA_TAG \
+                     -t $ECR_REGISTRY/chainlog-checksum-staging:$LATEST_TAG \
+                     -t $ECR_REGISTRY/chainlog-checksum-staging:$ENVIRONMENT_TAG \
+                     -f Dockerfile.checksum \
+                     .
+        docker push $ECR_REGISTRY/chainlog-checksum-staging --all-tags
+        # - chainlog-logger
+        docker pull $ECR_REGISTRY/chainlog-logger-staging:$LATEST_TAG || true
+        docker build -t chainlog-logger-staging \
+                     -t $ECR_REGISTRY/chainlog-logger-staging:$SHA_TAG \
+                     -t $ECR_REGISTRY/chainlog-logger-staging:$LATEST_TAG \
+                     -t $ECR_REGISTRY/chainlog-logger-staging:$ENVIRONMENT_TAG \
+                     -f Dockerfile.logger \
+                     .
+        docker push $ECR_REGISTRY/chainlog-logger-staging --all-tags
+
+    - name: Configure AWS credentials to connect to EKS cluster
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        aws-access-key-id: ${{ secrets.EKS_AWS_ACCESS_KEY_ID }}
```

Comment:
@OleksandrUA why do we need to sets of access key in same workflow?

Reply:
@cristidas `EKS_AWS_ACCESS_KEY_ID` is a K8s CICD user, it is allowed to connect to cluster and deploy containers there.
`AWS_ACCESS_KEY_ID` is a CICD user created by `app` TF module.
At some point we will decide to perform `terraform destroy` of all assets created by `app` module so I decided that from the very beginning we should have a separate CICD user for K8s deployments.

Reply:
@nickkounz once planning our K8s initiative team has decided that we won't implement any GitOps yet